index.js

```diff
@@ -1,30 +1,55 @@
 var fs = require("fs");
 var path = require("path");
-var detective = require('detective');
+var Walker = require('node-source-walk');
 var q = require('q');
 var walkdir = require("walkdir");
 var _ = require('lodash');
 var minimatch = require('minimatch');
 var util = require('util');
 
+function getArgumentFromCall(node) {
+  return node.type === 'CallExpression' && node.arguments[0]
+    ? node.arguments[0].value
+    : undefined;
+}
+
+function isRequireFunction(node) {
+  var callee = node.callee;
+  return callee && callee.type === 'Identifier' && callee.name === 'require' &&
+    getArgumentFromCall(node);
+}
+
+function isGruntLoadTaskCall(node) {
+  var callee = node.callee;
+  return callee && callee.property && callee.property.name === 'loadNpmTasks' &&
+    getArgumentFromCall(node);
+}
+
+function isImportDeclaration(node) {
+  return node.type === 'ImportDeclaration' && node.source && node.source.value;
+}
+
 function getModulesRequiredFromFilename(filename) {
   var content = fs.readFileSync(filename, "utf-8");
+  if (!content) {
+    return [];
+  }
+
+  var walker = new Walker();
```

Comment:
It could be a better choice to pass through the `options` object from depcheck to acorn. Just as [what node-source-walk package does](https://github.com/mrjoelkemp/node-source-walk/blob/master/index.js#L29). How do you think about this?